packages/plugin-js-packages/src/lib/runner/outdated/transform.ts

```diff
@@ -0,0 +1,102 @@
+import { Issue } from '@code-pushup/models';
+import { objectToEntries, pluralize } from '@code-pushup/utils';
+import { PackageDependency } from '../../config';
+import { outdatedSeverity } from './constants';
+import {
+  NormalizedOutdatedEntries,
+  NormalizedVersionOverview,
+  NpmOutdatedResultJson,
+  PackageVersion,
+  VersionType,
+} from './types';
+
+export function outdatedResultToAuditOutput(
+  result: NpmOutdatedResultJson,
+  dependenciesType: PackageDependency,
+) {
+  // current might be missing in some cases
+  // https://stackoverflow.com/questions/42267101/npm-outdated-command-shows-missing-in-current-version
+  const validDependencies: NormalizedOutdatedEntries = objectToEntries(result)
+    .filter(
+      (entry): entry is [string, NormalizedVersionOverview] =>
+        entry[1].current != null,
+    )
+    .filter(([, detail]) =>
+      dependenciesType === 'prod'
+        ? detail.type === 'dependencies'
+        : detail.type.startsWith(dependenciesType),
```

Comment:
Wouldn't this be safer?

```suggestion
        : detail.type === `${dependenciesType}Dependencies`,
```

Reply:
Not sure why safer but sure, I can do it this way.